libexec/klab-prove

```diff
@@ -52,41 +52,29 @@ fi
 
 echo "Proof ${bold}STARTING${reset}:" "$(basename "$target_spec")" "$dump_notice"
 
+logging=''
 {
 if [ ! -z "$QUIET" ]; then
-  mkdir -p $KLAB_OUT/log/
-  K_OPTS=-Xmx10G \
-    "$KLAB_EVMS_PATH/.build/k/k-distribution/target/release/k/bin/kprove" \
-    $dump_flags \
-    --directory "$KLAB_EVMS_PATH/.build/java/" \
-    --def-module RULES \
-    --output-omit "<programBytes> <program> <code> <previousGas> <touchedAccounts> <interimStates> <callStack> <callData>" \
-    --output-flatten "_Map_ #And" \
-    --no-alpha-renaming \
-    --no-sort-collections \
-    --output json \
-    --boundary-cells k,pc \
-    --smt_prelude $KLAB_OUT/prelude.smt2 \
-    --z3-tactic "(or-else (using-params smt :random-seed 3 :timeout 1000) (using-params smt :random-seed 2 :timeout 2000) (using-params smt :random-seed 1))" \
-    "$target_spec" > $KLAB_OUT/log/$spec_hash.log 2> $KLAB_OUT/log/$spec_hash.err.log
-  result=$?
-else
+    mkdir -p $KLAB_OUT/log/
+    logging='> $KLAB_OUT/log/$spec_hash.log 2> $KLAB_OUT/log/$spec_hash.err.log'
+fi
   K_OPTS=-Xmx10G \
     "$KLAB_EVMS_PATH/.build/k/k-distribution/target/release/k/bin/kprove" \
     $dump_flags \
     --directory "$KLAB_EVMS_PATH/.build/java/" \
     --def-module RULES \
-    --output-omit "<programBytes> <program> <code> <previousGas> <touchedAccounts> <interimStates> <callStack> <callData>" \
+    --output-omit "<programBytes> <program> <code> <previousGas> <touchedAccounts> <interimStates> <callStack> <callData> <substate> <previousHash> <ommersHash> <stateRoot> <transactionsRoot> <receiptsRoot> <logsBloom> <difficulty> <number> <gasLimit> <gasUsed> <extraData> <mixHash> <blockNonce> <ommerBlockHeaders> <blockhash> <txOrder> <txPending> <messages>" \
     --output-flatten "_Map_ #And" \
+    --z3-impl-timeout 300 --no-exc-wrap \
+    --cache-func-optimized --format-failures \
     --no-alpha-renaming \
     --no-sort-collections \
     --output json \
     --boundary-cells k,pc \
     --smt_prelude $KLAB_OUT/prelude.smt2 \
     --z3-tactic "(or-else (using-params smt :random-seed 3 :timeout 1000) (using-params smt :random-seed 2 :timeout 2000) (using-params smt :random-seed 1))" \
-    "$target_spec"
+    "$target_spec" $logging
```

Comment:
Have you tested this?
this is how we had this before but because `$logging` is stringified the `>` characters are being escaped and passed as parameters to `kprove` instead of being interpreted as shell instructions piping the stdout/stderr into files.

Reply:
Yeah, this works. The key is to have single quotes on the definition of `logging`

Reply:
the more you know